edit user form and implemented testing

justgarcia5 opened the pull request.

Commit:
edit user form and implemented testing

Commit:
added user feature and tests

JustinNealHatch committed:
Merge branch 'master' into features/edit-a-user-164704330

Commit:
updated features/user_spec test and passed

justgarcia5 committed:
Merge branch 'master' into features/edit-a-user-164704330

jasononaquest commented on app/views/users/edit.html.erb:
```diff
@@ -0,0 +1,34 @@
+  <h2>Edit User </h2>
+<div class='card full-width'>
+  <div class='w-form'>
+  <%= form_for @user do |f| %>
```
We should make use of the form partial here.

kevinyu05 replied:
form partial has been added

jasononaquest commented on app/views/users/index.html.erb:
```diff
@@ -4,17 +4,18 @@
+        <% if user.is_admin == true %>
+          <p class="init-is-admin">Admin</p>
+         <% end %>
+        <% if user.team_leader_id != nil and user.is_admin != true %>
```
I think these checks can be rewritten to not use `==` or `!=`. That is a generally more idiomatic approach when possible.

kevinyu05 replied:
Switched checks to be rewritten without == or !=

kevinyu05 committed:
added partial form, switched logic on users index

kevinyu05 committed:
fixed syntax

kevinyu05 committed:
fixed css on nav link & logo

jasononaquest commented on app/views/users/_users.html.erb:
```diff
@@ -0,0 +1,20 @@
+      <div class="card">
+        <h3 class="init-fullname"><%= user.fullname %></h3>
+          <p class="init-email"><%= user.email %></p>
+            <% if user.is_admin == true %>
```
This section of logic could use the same treatment as we have in index. I'm sorry for not noticing it sooner.

kevinyu05 replied:
Altered logic to represent same treatment in index. No worries!

jasononaquest reviewed (changes requested):
Fix the users partial to use similar logic to index and I think we'll be good to go.

kevinyu05 committed:
cleaned up logic to represent similar logic in index

hyosunko committed:
fixed error on edit user/initiative
add test for edit user/initiative
extract user form from users/index
Coauthored_by: Kevin Yu<kevinyu05@gmail.com>
Coauthered_by: Justin Garcia<justgarcia5@gmail.com>

jasononaquest reviewed (approved):
LGTM
